x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -63,55 +82,186 @@ export const installTransform = async (
+                transformModuleId,
+                installNameSuffix,
+                'template'
+              ),
+              template: mergedDestinationIndexTemplateInstallationWithMappings,
+            } as DestinationIndexTemplateInstallation);
+            packageAssets.set(
+              'destinationIndexTemplate',
+              mergedDestinationIndexTemplateInstallationWithMappings
+            );
+          } else {
+            packageAssets.set('destinationIndexTemplate', destinationIndexTemplate);
+
+            destinationIndexTemplates.push({
+              transformModuleId,
+              installationName: getTransformNameForInstallation(
+                installablePackage,
+                transformModuleId,
+                installNameSuffix,
+                'template'
+              ),
+              template: destinationIndexTemplate,
+              _meta: getESAssetMetadata({ packageName: installablePackage.name }),
+            } as DestinationIndexTemplateInstallation);
+          }
+        }
+      }
     });
 
-    installedTransforms = await Promise.all(installationPromises).then((results) => results.flat());
-  }
+    const indexTemplatesRefs = destinationIndexTemplates.map((template) => ({
+      id: template.installationName,
+      type: ElasticsearchAssetType.indexTemplate,
+    }));
+
+    const transformRefs = transforms.map((t) => ({
+      id: t.installationName,
+      type: ElasticsearchAssetType.transform,
+    }));
 
-  if (previousInstalledTransformEsAssets.length > 0) {
+    // get and save refs associated with the transforms before installing
     esReferences = await updateEsAssetReferences(
       savedObjectsClient,
       installablePackage.name,
       esReferences,
       {
+        assetsToAdd: [...indexTemplatesRefs, ...transformRefs],
         assetsToRemove: previousInstalledTransformEsAssets,
       }
     );
+
+    await Promise.all(
+      destinationIndexTemplates.map((destinationIndexTemplate) => {
+        return installComponentAndIndexTemplateForDataStream({
+          esClient,
+          logger,
+          componentTemplates: {},
+          indexTemplate: {
+            templateName: destinationIndexTemplate.installationName,
+            // @ts-expect-error Index template here should not contain data_stream property
+            // as this template is applied to only an index and not a data stream
+            indexTemplate: {
+              template: destinationIndexTemplate.template,
+              priority: 200,
+              index_patterns: [
+                transformsSpecifications
+                  .get(destinationIndexTemplate.transformModuleId)
+                  ?.get('destinationIndex').index,
+              ],
+              _meta: destinationIndexTemplate._meta,
+              composed_of: [],
+            },
+          },
+        });
+      })
+    );
+    // @TODO: Should we create indices for jobs that we are not starting automatically?
```

Comment:
Nit - assume this line won't be merged, but if it does, the term `jobs` should be replaced with `transforms`

Reply:
Thanks! Removed this comment after clarifying with Dave Roberts